pybop/costs/_likelihoods.py

```diff
@@ -32,6 +32,11 @@ def __init__(self, problem, sigma):
         self.sigma = None
         if sigma is not None:
             self.set_sigma(sigma)
+        else:
```

Comment:
I think this functionality is better suited within the `set_sigma` method, as there are similar checks implemented within that method.